apps/fz_http/lib/fz_http_web/live/user_live/show_live.ex

```diff
@@ -63,4 +63,59 @@ defmodule FzHttpWeb.UserLive.Show do
       end
     end
   end
+
+  @impl Phoenix.LiveView
+  def handle_event(action, %{"user_id" => user_id}, socket) when action in ~w(promote demote) do
+    if user_id == "#{socket.assigns.current_user.id}" do
+      {:noreply,
+       socket
+       |> put_flash(:error, "Changing your own role is not supported.")}
+    else
+      user = Users.get_user!(user_id)
+
+      role =
+        case action do
+          "promote" -> :admin
+          "demote" -> :unprivileged
+        end
+
+      case Users.update_user(user, %{role: role}) do
+        {:ok, user} ->
+          # Force reconnect with new role
+          FzHttpWeb.Endpoint.broadcast("users_socket:#{user.id}", "disconnect", %{})
+
+          {:noreply,
+           socket
+           |> assign(:user, user)
+           |> put_flash(:info, "User updated successfully.")}
+
+        {:error, changeset} ->
+          {:noreply,
+           socket
+           |> put_flash(
+             :error,
+             "Error updating user: #{ErrorHelpers.aggregated_errors(changeset)}"
+           )}
+      end
+    end
+  end
+
+  @action_and_message %{
```

Comment:
nit: Thoughts on keeping module attributes at the top of the file?

Reply:
For things that aren't affecting the whole module, and are kind of implementation details, I'd normally keep them where they are used. one example would be https://cs.github.com/elixir-ecto/ecto/blob/0d82133d4d643dcf7007e20dceafefae88453ab9/lib/ecto/query.ex#L1053

Reply:
Kk